feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.ts

```diff
@@ -168,6 +169,30 @@ export class ConfiguratorAddToCartButtonComponent implements OnInit, OnDestroy {
     );
   }
 
+  protected isQuoteCartActive(): Observable<boolean> {
+    return this.activeCartFacade
+      .getActive()
+      .pipe(map((cart) => cart.quoteCode !== undefined));
+  }
+
+  protected getTranslationKeyForAddToCart(
+    isAddToCart: boolean
+  ): Observable<string> {
+    let translationKey = of('configurator.addToCart.confirmation');
+    this.isQuoteCartActive()
+      .pipe(take(1))
```

Comment:
It's not needed to take, subscribe and use 'of' here! Please simply map the Observable<boolean> to Observable<string>!